p2p/security/noise/intermediate_transport.go

```diff
@@ -0,0 +1,50 @@
+package noise
+
+import (
+	"context"
+	"net"
+
+	"github.com/libp2p/go-libp2p-core/canonicallog"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/sec"
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+type NoiseOptions struct {
```

Comment:
Please avoid config structs and use the normal config function pattern.

Reply:
I considered that but want to avoid exposing the intermediate transport.